hedera-mirror-importer/src/main/resources/db/scripts/v2/migration.sh

```diff
@@ -147,18 +147,21 @@ getCursorConditions() {
   local pctPerCursor=$(echo "scale=10; 100 / ${ASYNC_TABLE_SPLITS}" | bc)
   local totalPct=0
   local -n cursors=$1 # Init array to hold cursor definitions
+  log "Will split data into ${pctPerCursor}% chunks"
   local stats=$(PGPASSWORD="${SOURCE_DB_PASSWORD}" psql -q --csv -t -h "${SOURCE_DB_HOST}" -d "${SOURCE_DB_NAME}" -p "${SOURCE_DB_PORT}" -U "${SOURCE_DB_USER}" -c "${query}")
-
   while IFS="," read -r pct_of_total lower_bound upper_bound; do
     upperBound=$((upper_bound > upperBound ? upper_bound : upperBound))
-    local groupRange=$((upperBound - lower_bound))
-    local rangeToTake=$(echo "scale=10; (${pctPerCursor} / 100) * ${groupRange}" | bc)
     totalPct=$(echo "(${totalPct} + ${pct_of_total})" | bc)
+    local groupRange=$((upperBound - lower_bound))
+    local rangeToTake=$(echo "scale=10; ${groupRange} / (${totalPct} / ${pctPerCursor})" | bc)
```

Comment:
nit: I believe this syntax might be a bit simpler/efficient:

```suggestion
    local rangeToTake=$(bc -S 10 -e "${groupRange} / (${totalPct} / ${pctPerCursor}")
```

Reply:
I can not make this syntax work and the man page doesn't specify these options.
```
NAME
       bc - An arbitrary precision calculator language

SYNTAX
       bc [ -hlwsqv ] [long-options] [  file ... ]

DESCRIPTION
       bc is a language that supports arbitrary precision numbers with interactive execution of statements.  There are some similarities in the syntax to the C programming language.  A standard math library is available by command line option.  If requested, the math library is defined before processing any files.  bc starts by processing code from all the files listed on the command line in the order listed.  After all files have been processed, bc reads from
       the standard input.  All code is executed as it is read.  (If a file contains a command to halt the processor, bc will never read from the standard input.)

       This version of bc contains several extensions beyond traditional bc implementations and the POSIX draft standard.  Command line options can cause these extensions to print a warning or to be rejected.  This document describes the language accepted by this processor.  Extensions will be identified as such.

   OPTIONS
       -h, --help
              Print the usage and exit.

       -i, --interactive
              Force interactive mode.

       -l, --mathlib
              Define the standard math library.

       -w, --warn
              Give warnings for extensions to POSIX bc.

       -s, --standard
              Process exactly the POSIX bc language.

       -q, --quiet
              Do not print the normal GNU bc welcome.

       -v, --version
              Print the version number and copyright and quit.

   NUMBERS
       The most basic element in bc is the number.  Numbers are arbitrary precision numbers.  This precision is both in the integer part and the fractional part.  All numbers are represented internally in decimal and all computation is done in decimal.  (This version truncates results from divide and multiply operations.)  There are two attributes of numbers, the length and the scale.  The length is the total number of decimal digits used by bc to represent  a
       number and the scale is the total number of decimal digits after the decimal point.  For example:
               .000001 has a length of 6 and scale of 6.
               1935.000 has a length of 7 and a scale of 3.
```

Reply:
Probably macos syntax differs from linux.